docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
 # Developer Guide
 
-## Acknowledgements
+## Design
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{To Be Added}
 
-## Design & implementation
+---
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Implementation
+
+### Add feature
 
+The add feature is implemented using a `AddCommand` class extending from `Command` Class. 
+`AddCommand` contains an `execute` methods, which utilises the internal `spiltDetails` methods to extract the food details
+and to construct new `food` object .
+It is also facilitated by `Foodlist` Class, using `addFood` method of `FoodList` Class.
 
-## Product scope
-### Target user profile
+The add feature implementation is as follows:
+1. New AddCommand object is created by passing in a String containing food details
+2. `splitDetails` is called to split the string
+3. boolean `hasQuantity` is used to record whether the string contains `-q` quantity flag
+4. Split the string using ` -q ` regardless of whether the string contains the quantity part
+5. Check the index of `-n` and `-e` and split the string to save the details of the food name and expiry date
+6. If `hasQuantity` is `true`, return a String array with `name`, `date` and `quantity`, else only return
+a String array with `name` and `date`.
+7. Create a new `Food` object and use `addFood` to add the new food into the food list
+8. Return a `CommandResult` to show the successful message to the user
 
-{Describe the target user profile}
 
-### Value proposition
+**Class Diagram**
+
+![ClassDiagram](images/AddCommandClassDiagram.png)
```

Comment:
Is this format correct for UML diagram? Should it be +, - for the attributes and methods instead?
![image](https://user-images.githubusercontent.com/88440925/228425519-39d43f68-7461-4c07-b2df-dd3c5616becf.png)

Reply:
Is this format correct for UML diagram? Should it be {abstract} or <<interface>> for the command class that others classes extend from?
![image](https://user-images.githubusercontent.com/88440925/228425692-12eb8f0f-ed77-473f-96e1-3b760def22c0.png)

Reply:
Should change the notation

Reply:
The "C" icon may not be consistent with module standards, perhaps you could revise this?